### ISSUE TITLE
Binding socket to an occupied port doesn't fail

### ISSUE THREAD

**Issue by jbatnozic** [zerotier/libzt#111]
Basically what the title says. I compiled the example "server.cpp", and tried to run it twice (at the same time) on the same port to see what would happen. No error. However, when sending traffic to that port, only the first server would receive it.

This is very inconvenient; the user might wonder why his connection isn't working out but there is no error, when in fact the port is already taken.

I compiled the example on Windows with Visual Studio 2019.

Note: If you cannot reproduce the issue, try changing the socket type to ZTS_SOCK_DGRAM (you will have to delete or comment out the code after binding because acceptors are tcp-specific), and try again.

**Comment by joseph-henry** [comment]
So, a couple of things. The port you bind or connect to using `zts_` calls only really matters to that specific *app instance* since it has its own network stack. The behavior you're observing (I think) is that you are using the same port twice (this is ok!), but that you're running two instances of libzt possibly on the same `service port` or possibly even the same identities (which is definitely not allowed).

tl;dr: libzt needs a `service port` just like the ZeroTier daemon's 9993. And the identities must be unique otherwise you'll see undefined behavior like the above.

If I got something wrong please correct me and I can do some testing.
